virttest/utils_test/libvirt.py

```diff
@@ -2371,19 +2372,46 @@ def create_tpm_dev(params):
     return tpm_dev
 
 
-def get_vm_device(vmxml, dev_tag, index=0):
+def get_vm_device(vmxml, dev_tag, dev_dict={}, index=0):
     """
-    Get current vm device according to device tag
+    Get current vm device according to device tag,
+    if no this device , add one device
 
     :param vmxml: domain VMXML instance
     :param dev_tag: device tag
+    :param dev_dict: dict to create device
     :param index: device index
     :return: device object
     """
     xml_devices = vmxml.devices
-    dev_index = xml_devices.index(xml_devices.by_device_tag(dev_tag)[index])
-    dev_obj = xml_devices[dev_index]
-    return (dev_obj, xml_devices)
+    try:
+        dev_index = xml_devices.index(xml_devices.by_device_tag(dev_tag)[index])
+        dev_obj = xml_devices[dev_index]
+        return (dev_obj, xml_devices)
+
+    except IndexError:
+        LOG.debug('Add one new :%s', dev_tag)
+        new_device = ''
+        if dev_tag == 'disk':
+            new_device = disk.Disk()
+        elif dev_tag == 'controller':
+            new_device = controller.Controller()
+        elif dev_tag == 'interface':
+            new_device = interface.Interface()
+        elif dev_tag == 'channel':
+            new_device = channel.Channel()
+        elif dev_tag == 'vsock':
+            new_device = vsock.Vsock()
+        elif dev_tag == 'watchdog':
+            new_device = watchdog.Watchdog()
+        elif dev_tag == 'rng':
+            new_device = rng.Rng()
+        elif dev_tag == 'hostdev':
+            new_device = hostdev.Hostdev()
+
+        new_device.setup_attrs(**dev_dict)
+        add_vm_device(vmxml, new_device)
+        return (new_device, xml_devices)
```

Comment:
In the original codes, I do not know why we return all devices by `xml_devices`. However, I do not find any reference for this function in other codes. So should we only return one device object?

Reply:
yes , there is not any reference for this function in other codes.
I removed it `xml_devices`
BTW: Could you please help confirm which device needs to be added in` if else branch` or these is enough , because  there are many devices under the folder of  ` ./virttest/libvirt_xml/devices/*` @dzhengfy

Reply:
So far we do not need to add all devices. In future if anyone want to use, it is so easy to extend this function as well as test result .